darwin/dataset/upload_manager.py

```diff
@@ -44,6 +44,35 @@ def full_path(self):
         return construct_full_path(self.data["path"], self.data["filename"])
 
 
+class FileMonitor(object):
+    """
+    An object used to monitor the progress of a :class:`BufferedReader`.
+
+    To use this monitor, you construct your :class:`BufferedReader` as you
+    normally would, then construct this object with it as argument.
+    """
+
+    def __init__(self, io: BinaryIO, file_size: int, callback):
+        #: Instance of the :class:`BufferedReader` being monitored
+        self.io = io
+
+        #: Optionally function to call after a read
+        self.callback = callback
+
+        #: Number of bytes already read from the :class:`BufferedReader`
```

Comment:
I'm not really a python dev, so I may have some weird biases as i developed in this language so little. 
Here, this is just a habbit of mine that may not make sense here - `":` comments are special comments that are picked up by `Sphinx` when generating docs.

Reply:
Here we prefer the rst documentation style instead of the `#:`. Please see our Python guidelines for an example:

 - https://www.notion.so/v7labs/Python-Code-Style-351a7831c2074e78bf4e21ec13777542